pandas/core/internals/blocks.py

```diff
@@ -1752,33 +1757,6 @@ def is_view(self) -> bool:
     def is_numeric(self):
         return self.values.dtype._is_numeric
 
-    def take_nd(
-        self,
-        indexer: npt.NDArray[np.intp],
-        axis: int = 0,
-        new_mgr_locs: BlockPlacement | None = None,
-        fill_value=lib.no_default,
-    ) -> Block:
-        """
-        Take values according to indexer and return them as a block.
-        """
-        if fill_value is lib.no_default:
-            fill_value = None
-
-        # TODO(EA2D): special case not needed with 2D EAs
-        # axis doesn't matter; we are really a single-dim object
-        # but are passed the axis depending on the calling routing
-        # if its REALLY axis 0, then this will be a reindex and not a take
-        new_values = self.values.take(indexer, fill_value=fill_value, allow_fill=True)
```

Comment:
So `fill_value=None/allow_fill=True` is essentially equivalent to `fill_value=self.fill_value/allow_fill=False` (in Block.take_nd)?

Reply:
AFAICT in all the relevant cases (definitely all the existing test cases) allow_fill should be True iff indexer contains -1s, i.e. it is benign to set it to always-True